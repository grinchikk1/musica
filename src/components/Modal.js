import { Component } from "react";
import PropTypes from "prop-types";
import "../styles/Modal.scss";

class Modal extends Component {
  render() {
    const { header, closeButton, text, actions, isOpen, onRequestClose } =
      this.props;

    if (!isOpen) {
      return null;
    }

    return (
      <div className="modal-overlay" onClick={onRequestClose}>
        <div className="modal" onClick={(e) => e.stopPropagation()}>
          <div className="modal-header">
            <h2>{header}</h2>
            {closeButton && (
              <button className="modal-close" onClick={onRequestClose}>
                &times;
              </button>
            )}
          </div>
          <div className="modal-content">{text}</div>
          <div className="modal-actions">{actions}</div>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  header: PropTypes.string.isRequired,
  closeButton: PropTypes.bool.isRequired,
  text: PropTypes.string.isRequired,
  actions: PropTypes.element.isRequired,
  isOpen: PropTypes.bool.isRequired,
  onRequestClose: PropTypes.func.isRequired,
};

export default Modal;
