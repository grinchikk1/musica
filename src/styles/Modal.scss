.modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 999;
}

.modal {
  position: relative;
  background-color: #fff;
  border-radius: 5px;
  width: 600px;

  h2 {
    margin: 0;
    font-size: 18px;
  }
}

.modal-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-radius: 5px 5px 0 0;
  padding: 20px;
  background: ghostwhite;
}

.modal-close {
  border: none;
  color: inherit;
  background-color: transparent;
  font-size: 30px;
  cursor: pointer;

  &:hover {
    opacity: 0.7;
  }
}

.modal-content {
  text-align: center;
  padding: 30px 100px;
  line-height: 26px;
  font-size: 14px;
}

.modal-actions {
  display: flex;
  justify-content: center;
  padding: 20px;

  button {
    margin-left: 10px;
  }
}
