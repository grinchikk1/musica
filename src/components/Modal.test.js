import React from "react";
import renderer from "react-test-renderer";
import Modal from "./Modal";

describe("Modal Component", () => {
  // snapshot test
  it("renders correctly when isOpen is true", () => {
    const tree = renderer
      .create(
        <Modal
          header="Modal Header"
          closeButton={true}
          text="Modal Content"
          actions={<button>Close</button>}
          isOpen={true}
          onRequestClose={() => {}}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
