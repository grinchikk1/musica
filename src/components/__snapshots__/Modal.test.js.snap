// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders correctly when isOpen is false 1`] = `null`;

exports[`Modal Component renders correctly when isOpen is true 1`] = `
<div
  className="modal-overlay"
  onClick={[Function]}
>
  <div
    className="modal"
    onClick={[Function]}
  >
    <div
      className="modal-header"
    >
      <h2>
        Modal Header
      </h2>
      <button
        className="modal-close"
        onClick={[Function]}
      >
        Ã—
      </button>
    </div>
    <div
      className="modal-content"
    >
      Modal Content
    </div>
    <div
      className="modal-actions"
    >
      <button>
        Close
      </button>
    </div>
  </div>
</div>
`;
